src/silo/query_engine/query.cpp

```diff
@@ -38,23 +38,22 @@ QueryPlan Query::toQueryPlan(
 ) const {
    SPDLOG_DEBUG("Parsed filter: {}", filter->toString());
 
-   auto partition_filter_operators = std::make_shared<filter::operators::OperatorVector>();
-   partition_filter_operators->reserve(database->table->getNumberOfPartitions());
+   std::vector<CopyOnWriteBitmap> partition_filters;
+   partition_filters.reserve(database->table->getNumberOfPartitions());
    for (size_t partition_index = 0; partition_index < database->table->getNumberOfPartitions();
         partition_index++) {
-      partition_filter_operators->emplace_back(filter->compile(
+      auto filter_operator = filter->compile(
          *database,
          database->table->getPartition(partition_index),
          filter::expressions::Expression::AmbiguityMode::NONE
-      ));
+      );
       SPDLOG_DEBUG(
-         "Simplified query for partition {}: {}",
-         partition_index,
-         partition_filter_operators->back()->toString()
+         "Simplified query for partition {}: {}", partition_index, filter_operator->toString()
       );
+      partition_filters.emplace_back(filter_operator->evaluate());
```

Comment:
this is the main change in the code, evaluate is now called during the `toQueryPlan` function and not afterwards. This means that errors will be seen before creating the header (header is created here: https://github.com/GenSpectrum/LAPIS-SILO/blob/main/src/silo/api/query_handler.cpp#L45)